AmplifyTests/CategoryTests/DataStore/Model/LazyReferenceTests.swift

```diff
@@ -84,4 +84,23 @@ final class LazyReferenceTests: XCTestCase {
         let json = try comment.toJSON()
         XCTAssertEqual(json, "{\"post\":[{\"name\":\"id\",\"value\":\"postId\"}],\"id\":\"commentId\",\"content\":\"content\",\"updatedAt\":null,\"createdAt\":null}")
     }
+    
+    func testDecodePrimaryKeysOnly() async throws {
+        let postIdentifierName = "id"
+        let postIdentifierValue = "postId"
+        
+        let commentWithLazyPostJSON = "{\"post\":{\"\(postIdentifierName)\":\"\(postIdentifierValue)\"},\"id\":\"commentId\",\"content\":\"c\",\"updatedAt\":null,\"createdAt\":null}"
+        
+        guard let decodedComment = try ModelRegistry.decode(modelName: LazyChildComment4V2.modelName, from: commentWithLazyPostJSON) as? LazyChildComment4V2 else {
+            XCTFail("Could not decode to comment from json")
+            return
+        }
+        switch decodedComment._post.loadedState {
+        case .notLoaded(let identifiers):
+            let id = identifiers?.first(where: { $0.name == postIdentifierName && $0.value == postIdentifierValue })
+            XCTAssertNotNil(id)
```

Comment:
This is good, but how about we assert there's nothing unexpected in the identifiers array? Maybe something like:

```
let identifiers = try XCTUnwrap(unloadedIdentifiers)
let descriptions = identifiers.map { "\($0.name): \($0.value)" }
XCTAssertEqual(descriptions, ["\(postIdentifierName): \(postIdentifierValue)"])
```